services/web/server/src/simcore_service_webserver/login/_2fa.py

```diff
@@ -0,0 +1,109 @@
+""" two-factor-authentication utils
+
+Currently includes two parts:
+
+- generation and storage of secret codes for 2FA validation (using redis)
+- sending SMS of generated codes for validation (using twilio service)
+
+"""
+
+import asyncio
+import logging
+import secrets
+from typing import Optional
+
+from aiohttp import web
+from pydantic import BaseModel, Field
+from servicelib.logging_utils import log_decorator
+from twilio.rest import Client
+
+from ..redis import get_redis_validation_code_client
+from .settings import TwilioSettings
+
+log = logging.getLogger(__name__)
+
+
+class ValidationCode(BaseModel):
+    value: str = Field(..., description="The code")
+
+
+#
+# REDIS:
+#  generation, storage of secret codes
+#
```

Comment:
maybe make it a bit more meaningful or remove this comment. I did not find it that helpful